### PR TITLE
Ft add module bucket creation

### PR DESCRIPTION
**Pull request by Fab-T**
Creating a module for creating buckets.

Use case: Broad-aou-cromwell

Single bucket creation with ACLs bindings.

### CODE REVIEW

File: terraform-modules/storage-bucket/bucket.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+terraform {
```
Comment (hjfbynara) [inline terraform-modules/storage-bucket/bucket.tf:1]: this should go in the versions.tf file.  do you need to specify a patch version or is it ok just to be >= 0.12.  versions.tf says >= 0.12.  if 0.12.6 is min supported then update versions.tf with that.  still remove it from here.

I realize that you copied this from another module but I think it is confusing to have these spread out in diff parts of module.  the entire module should just specify the min version any resource it creates needs.

File: terraform-modules/storage-bucket/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+```
+### Variables to set
+```
+# name of bucket
+bucket_name =
+# name of provider
+provider =
+# ACL list
+bindings =
+
```
Comment (hjfbynara) [inline terraform-modules/storage-bucket/README.md:45]: location and storage_class are also two variables that you can pass in so should be in README

File: terraform-modules/storage-bucket/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
+output "bucket" {
```
Comment (hjfbynara) [inline terraform-modules/storage-bucket/outputs.tf:1]: I am pretty sure if you specify the top-level resource then you have access to all the lower level variable that make up the resource.  so name and url I don't think are necessary as separate outputs

File: terraform-modules/storage-bucket/versions.tf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+terraform {
+  required_version = ">= 0.12"
```
Comment (hjfbynara) [inline terraform-modules/storage-bucket/versions.tf:2]: so make this 0.12.6 and remove from other files

File: terraform-modules/storage-bucket/README.md (hunk 2 above)
Comment (Fab-T) [reply terraform-modules/storage-bucket/README.md:45]: yes actually I just wanted to highlight the required ones , but I noticed your formatting using a table and I will use the same

File: terraform-modules/storage-bucket/outputs.tf (hunk 3 above)
Comment (Fab-T) [reply terraform-modules/storage-bucket/outputs.tf:1]: yep ,  I actually reused the output from another module, will remove it

**Comment by Fab-T** [comment]
Added - lifecycle and retention
Note: logging is not possible at creation time.

Bucket creation tested on gotc-dev
Tested with retention and versioning = false
Tested with lifecycle

File: terraform-modules/storage-bucket/bucket.tf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  dynamic "retention_policy" {
+    for_each = var.retention_policy == null ? [] : [var.retention_policy]
+    content {
+      is_locked        = var.retention_policy.is_locked
+      retention_period = var.retention_policy.retention_period
+    }
+  }
+
+  dynamic "lifecycle_rule" {
```
Comment (hjfbynara) [inline terraform-modules/storage-bucket/bucket.tf:22]: not sure from the code but is the default mode to have no lifecycle policy set.  so if I don't provide a lifecycle policy when I call the module - then lifecycle is not enabled on bucket.

Comment (hjfbynara) [reply terraform-modules/storage-bucket/bucket.tf:22]: to further clarify does providing a empty map to terraform for lifecycle rule - mean lifecycle is disabled.  or is it enabled but with a do nothing clause?  easy to test - create a bucket not specifying lifecycle and then run gsutil lifecycle get on bucket.  that will return if it is enabled or not and what the rule is.

Comment (Fab-T) [reply terraform-modules/storage-bucket/bucket.tf:22]: I confirm when I am testing the terraform plan output provides this result with no lifecycle  block declared : 

# module.aou-in-out-bucket.google_storage_bucket.bucket will be created
  + resource "google_storage_bucket" "bucket" {
      + bucket_policy_only = (known after apply)
      + force_destroy      = false
      + id                 = (known after apply)
      + location           = "US"
      + name               = "aou-input"
      + project            = (known after apply)
      + self_link          = (known after apply)
      + storage_class      = "MULTI_REGIONAL"
      + url                = (known after apply)

      + versioning {
          + enabled = true
        }
    }

And when I ran the command : 
wm745-040:test ftraquet$ gsutil lifecycle get gs://aou-input
gs://aou-input/ has no lifecycle configuration.

Comment (Fab-T) [reply terraform-modules/storage-bucket/bucket.tf:22]: I then updated my test.tf file to include lifecycle and then updated the bucket.
Lifecycle is present 
wm745-040:test ftraquet$ gsutil lifecycle get gs://aou-input
gs://aou-input/ has no lifecycle configuration.

**Pull request merged**